app/bundles/CampaignBundle/Controller/ImportController.php

```diff
@@ -0,0 +1,280 @@
+            return $this->accessDenied();
+        }
+
+        $forceStop = $request->get('cancel', false);
+        $step      = ($forceStop) ? self::STEP_UPLOAD_ZIP : $this->requestStack->getSession()->get('mautic.campaign.import.step', self::STEP_UPLOAD_ZIP);
+        $importDir = $this->getImportDirName();
+        $fileName  = $this->getImportFileName();
+        $fs        = new Filesystem();
+        $complete  = false;
+
+        $fullPath  = $this->getFullZipPath();
+
+        if ($ignorePost) {
+            dump($step, $fileName, $fullPath);
+        }
+
+        if (!file_exists($fullPath) && self::STEP_UPLOAD_ZIP !== $step) {
+            // Force step one if the file doesn't exist
+            $this->logger->log(LogLevel::WARNING, "File {$fullPath} does not exist anymore. Reseting import to step STEP_UPLOAD_ZIP.");
+            $this->addFlashMessage('mautic.import.file.missing', ['%file%' => $fullPath], FlashBag::LEVEL_ERROR);
+            $step = self::STEP_UPLOAD_ZIP;
+            $this->requestStack->getSession()->set('mautic.campaign.import.step', self::STEP_UPLOAD_ZIP);
+        }
+
+        $progress = (new Progress())->bindArray($this->requestStack->getSession()->get('mautic.campaign.import.progress', [0, 0]));
+        $action   = $this->generateUrl('mautic_campaign_import_action', ['objectAction' => 'new']);
+
+        switch ($step) {
+            case self::STEP_UPLOAD_ZIP:
+                if ($forceStop) {
+                    $this->resetImport();
+                    $this->removeImportFile($fullPath);
+                    $this->logger->log(LogLevel::WARNING, "Import for file {$fullPath} was force-stopped.");
+                }
+
+                $form = $this->formFactory->create(CampaignImportType::class, [], ['action' => $action]);
+                break;
+            case self::STEP_PROGRESS_BAR:
+                // Just show the progress form
+                $this->requestStack->getSession()->set('mautic.campaign.import.step', self::STEP_IMPORT_FROM_ZIP);
+                break;
+        }
+
+        if (!$ignorePost && 'POST' === $request->getMethod()) {
+            if (!isset($form) || $this->isFormCancelled($form)) {
+                $this->resetImport();
+                $this->removeImportFile($fullPath);
+                $reason = isset($form) ? 'the form is empty' : 'the form was canceled';
+                $this->logger->log(LogLevel::WARNING, "Import for file {$fullPath} was aborted because {$reason}.");
+
+                return $this->newAction($request, true);
+            }
+
+            if (self::STEP_UPLOAD_ZIP === $step) {
+                $valid = $this->isFormValid($form);
+
+                if ($valid) {
+                    if (file_exists($fullPath)) {
+                        unlink($fullPath);
+                    }
+
+                    $fileData = $request->files->get('campaign_import')['campaignFile'] ?? null;
```

Comment:
Can we call it entity_import and entityFile? Like if someone would like to import another entity like just an email or landing page then it would be possible?

Looking also at the location, perhaps we could move this to the core bundle as it could be used by all the bundles and plugins as well?